pkg/storer/reserve.go

```diff
@@ -276,21 +281,29 @@ func (db *DB) removeChunk(ctx context.Context, address swarm.Address, batchID []
 
 func (db *DB) reserveCleanup(ctx context.Context) error {
 	dur := captureDuration(time.Now())
+	removed := 0
 	defer func() {
 		db.metrics.MethodCallsDuration.WithLabelValues("reserve", "cleanup").Observe(dur())
+		db.metrics.ReserveSize.Set(float64(db.reserve.Size()))
```

Comment:
what about the Reserve.Size ?
We set the metric but we are not adjusting the size of the reserve. Also there is no easy way to do it as we are iterating with bin 0. So there could be chunks which are not accounted for in the reserve size. We will need to do a complete recalculation after the cleanup is done I think.

Reply:
This is happening right now. After restart the reserve values drop.

![Screenshot 2023-06-28 at 11 54 01 PM](https://github.com/ethersphere/bee/assets/5412421/2d4b46ce-c2bc-4920-a536-8bd65c913ab3)

Reply:
> what about the Reserve.Size ?

He's right.  I missed the calls to db.reserve.AddSize(1) and db.reserve.AddSize(-evicted) in ReservePutter and evictBatch above.

Reply:
nice catch